python_modules/dagster/dagster/_generate/generate.py

```diff
@@ -120,23 +125,46 @@ def _generate_files_from_template(
                 f.write("\n")
 
 
+def generate_project_from_example(path: str, example: str):
+
+    EXAMPLE_WHITELIST = ["docs_snippets"]
+
+    normalized_path = os.path.normpath(path)
+
+    example_folder = file_relative_path(__file__, "../../../../examples")
+    example_list = [
+        e
+        for e in os.listdir(example_folder)
+        if (e not in EXAMPLE_WHITELIST and not _should_skip_file(e))
+    ]
+
+    if example not in example_list:
+        click.echo(
+            click.style(f'Can\'t find example "{example}". ', fg="red")
+            + "\nPlease specify one of the official Dagster examples. "
+            + "You can find the examples in the Dagster repo: https://github.com/dagster-io/dagster/tree/master/examples"
+        )
+        sys.exit(1)
+
+    click.echo(f"Creating a Dagster project at {normalized_path}.")
+
+    selected_example_path = os.path.join(example_folder, example)
+    shutil.copytree(
+        src=selected_example_path,
+        dst=normalized_path,
+        ignore=shutil.ignore_patterns(*IGNORE_PATTERN_LIST),
+    )
```

Comment:
ok.. this won't work because site-packages won't have files outside dagster/ dir
```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/yuhan/.pyenv/versions/3.8.6/envs/aaa/lib/python3.8/site-packages/dagster/_generate/../../../../examples'
```